R/DataPulls.R

```diff
@@ -0,0 +1,225 @@
+        targetId = targetId,
+        comparatorId = comparatorId,
+        databaseId = databaseId,
+        analysisId = analysisId
+      )
+    sql <-
+      SqlRender::translate(sql = sql, targetDialect = connection@dbms)
+    result <-
+      DatabaseConnector::querySql(
+        connection = connection,
+        sql = sql,
+        snakeCaseToCamelCase = TRUE
+      )
+  } else {
+    result <- preferenceScoreDist %>%
+      dplyr::filter(
+        targetId %in% !!targetId,
+        comparatorId %in% !!comparatorId,
+        databaseId %in% !!databaseId,
+        analysisId %in% !!analysisId
+      )
+  }
+  return(result)
+}
+
+
+.getExposureOfInterest <- function(connectionDetails = NULL,
+                                   exposureOfInterest = NULL,
+                                   databaseSchema = NULL,
+                                   targetId ,
+                                   comparatorId) {
+  if (!is.null(connectionDetails)) {
+    connection <- DatabaseConnector::connect(connectionDetails)
+    sql <-   "SELECT *
+              FROM @databaseSchema.exposure_of_interest
+              WHERE exposure_id IN (@targetId, @comparatorId);"
+    
+    sql <-
+      SqlRender::render(
+        sql = sql,
+        databaseSchema = databaseSchema,
+        targetId = targetId,
+        comparatorId = comparatorId
+      )
+    sql <-
+      SqlRender::translate(sql = sql,
+                           targetDialect = connection@dbms)
+    result <-
+      DatabaseConnector::querySql(
+        connection = connection,
+        sql = sql,
+        snakeCaseToCamelCase = TRUE
+      )
+  } else {
+    result <- exposureOfInterest %>%
+      dplyr::filter(exposureId %in% c(targetId, comparatorId))
+  }
+  return(result)
+}
+
+#' Get the preference score distribution for plotting
+#'
+#' @details
+#' Get preference score data from data stored in OHDSI comparative effectiveness data model,
+#' The output of this function  maybe used to create plots or tables
+#' that compares the preference score distribution between target and comparators.
+#'
+#' @template optionalConnectionDetails
+#' @template optionalDatabaseSchema
+#' @template targetId
+#' @template comparatorId
+#' @template databaseId
+#' @template analysisId
+#' @param preferenceScoreDist  (optional) A R-dataFrame object with fields named in camelCase containing
+#'                             data from preference_score_dist table in comparative effectiveness data model.
+#' @param exposureOfInterest   (optional) A R-dataFrame object with fields named in camelCase containing
+#'                             data from exposure_of_interest table in comparative effectiveness data model.
+#' @return                     Tibble with fields preference score, density, cohort-name,
+#'                             group for identifying if the cohort is target vs comparator.
+#'
+#' @examples
+#' # example1: If the comparative effectiveness results are stored in a 
```

Comment:
This is a lot of text for comments in the example code. I would move this to the description section of this function.

In the example code, I literally would just say '#If you're data is in a database' above the one function call, and '#If you're data is in data frames' above the other.